### PR TITLE
Add ES|QL async security tests

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit expands the current ESQL security tests to cover async.

relates #102455

### CODE REVIEW

File: x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java [hunk 1]
```diff
@@ -293,19 +314,6 @@ private Response runESQLCommand(String user, String command) throws IOException
...
-        query.startObject();
-        query.field("query", command);
-        if (pragmas != Settings.EMPTY) {
-            query.startObject("pragma");
-            query.value(pragmas);
-            query.endObject();
-        }
-        query.endObject();
-        Request request = new Request("POST", "_query");
-        request.setJsonEntity("{\"query\":\"" + command + "\"}");
```
Comment (ChrisHegarty) [inline x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java:306]: Looks like pragmas were never previously sent in this test. I fixed this for the non-async case as part of the test refactoring.

File: x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java [hunk 2]
```diff
@@ -271,11 +271,32 @@ private void removeEnrichPolicy() throws Exception {
...
+        json.endObject();
+        Request request = new Request("POST", "_query");
+        request.setJsonEntity(Strings.toString(json));
+        request.setOptions(RequestOptions.DEFAULT.toBuilder().addHeader("es-security-runas-user", user));
+        return client().performRequest(request);
+    }
+
+    static void addRandomPragmas(XContentBuilder builder) throws IOException {
+        Settings pragmas = randomPragmas();
+        if (pragmas != Settings.EMPTY) {
```
Comment (nik9000) [inline x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java:292]: I imagine we have to skip this if we're running in a release build.

**Review by nik9000 — APPROVED** [review]

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks Chris!

File: x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java (hunk 2 above)
Comment (ChrisHegarty) [reply x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java:292]: Oh yes. I inadvertently removed this. I’ll add it back. Thanks.

**Review by bpintea — APPROVED** [review]

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Pull request merged**